src/core/metallicroughnesseffect.cpp

```diff
@@ -718,8 +719,25 @@ void MetallicRoughnessEffect::setToneMappingAlgorithm(MetallicRoughnessEffect::T
     emit toneMappingAlgorithmChanged(algorithm);
 }
 
+Qt3DRender::QAbstractTexture *MetallicRoughnessEffect::brdfLUT() const
+{
+    return m_brdfLUTParameter->value().value<Qt3DRender::QAbstractTexture *>();
+}
+
 void MetallicRoughnessEffect::setBrdfLUT(Qt3DRender::QAbstractTexture *brdfLUT)
 {
+    auto d = Qt3DCore::QNodePrivate::get(this);
+    auto current = this->brdfLUT();
+    if (current) {
+        d->unregisterDestructionHelper(current);
+        if (current->parentNode() == this)
+            current->setParent(static_cast<Qt3DCore::QNode *>(nullptr));
+    }
+    if (brdfLUT) {
+        d->registerDestructionHelper(brdfLUT, &MetallicRoughnessEffect::setBrdfLUT, brdfLUT);
+        if (!brdfLUT->parentNode())
+            brdfLUT->setParent(this);
```

Comment:
Usually it's just  if (!brdfLUT->parent()) brdfLUT->setParent(this); In practice I would assume parentNode() == parent() but might be better to align the code with what we do everywhere else.

Reply:
done, although it's kind of odd since the setParent MUST the one that takes a node, the other one being deleted...